fix: display issue with long equations

Hasenpfote opened the pull request:
Fix for Long Equations Cut Off (#285)

I fixed the issue where long equations were being cut off in #285. Submitting on behalf of xiyuvi.

Hasenpfote commented:
In my new Vercel environment, there are no errors in the [logs](https://github.com/saicaca/fuwari/issues/285#issuecomment-2594406510).
However, it seems there are issues here that we cannot verify, as xiyuvi mentioned.
Could someone please check the situation?

L4Ph commented:
Please provide the following:

- Node.js version
- pnpm version
- Local build results

This has happened with my previous PRs as well, but the build sometimes crashes randomly.

Hasenpfote commented:
This is my first time using Vercel, so it took me a bit longer.

Node.js version: 22.x (current default)
pnpm version: 9.x (current default)
Local build results:
```shell
[12:04:52.011] Running build in Washington, D.C., USA (East) – iad1
[12:04:52.709] Cloning github.com/Hasenpfote/fuwari (Branch: fix-equation-cutoff, Commit: 48b9c13)
[12:04:52.863] Skipping build cache, deployment was triggered without cache.
[12:04:53.282] Cloning completed: 575.21ms
[12:04:53.671] Running "vercel build"
[12:04:54.130] Vercel CLI 39.3.0
[12:04:55.107] Detected `pnpm-lock.yaml` version 9 generated by pnpm@9.x
[12:04:55.125] Installing dependencies...
[12:04:57.389] Lockfile is up to date, resolution step is skipped
[12:04:57.503] Progress: resolved 1, reused 0, downloaded 0, added 0
[12:04:57.697] Packages: +1040
[12:04:57.697] ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
[12:04:58.504] Progress: resolved 1040, reused 0, downloaded 16, added 4
[12:04:59.504] Progress: resolved 1040, reused 0, downloaded 76, added 62
[12:05:00.506] Progress: resolved 1040, reused 0, downloaded 133, added 120
[12:05:01.508] Progress: resolved 1040, reused 0, downloaded 187, added 171
[12:05:02.508] Progress: resolved 1040, reused 0, downloaded 300, added 291
[12:05:03.510] Progress: resolved 1040, reused 0, downloaded 423, added 413
[12:05:04.513] Progress: resolved 1040, reused 0, downloaded 503, added 503
[12:05:05.514] Progress: resolved 1040, reused 0, downloaded 661, added 647
[12:05:06.528] Progress: resolved 1040, reused 0, downloaded 722, added 708
[12:05:08.666] Progress: resolved 1040, reused 0, downloaded 722, added 709
[12:05:09.666] Progress: resolved 1040, reused 0, downloaded 763, added 755
[12:05:10.668] Progress: resolved 1040, reused 0, downloaded 845, added 839
[12:05:11.668] Progress: resolved 1040, reused 0, downloaded 937, added 924
[12:05:12.670] Progress: resolved 1040, reused 0, downloaded 980, added 965
[12:05:13.671] Progress: resolved 1040, reused 0, downloaded 1039, added 1033
[12:05:13.688] Progress: resolved 1040, reused 0, downloaded 1040, added 1040, done
[12:05:14.065] .../node_modules/@parcel/watcher install$ node scripts/build-from-source.js
[12:05:14.066] .../sharp@0.33.5/node_modules/sharp install$ node install/check
[12:05:14.067] .../esbuild@0.21.5/node_modules/esbuild postinstall$ node install.js
[12:05:14.067] .../.pnpm/swup@4.8.1/node_modules/swup postinstall$ opencollective-postinstall || true
[12:05:14.091] .../node_modules/@biomejs/biome postinstall$ node scripts/postinstall.js
[12:05:14.191] .../node_modules/@parcel/watcher install: Done
[12:05:14.265] .../.pnpm/swup@4.8.1/node_modules/swup postinstall: Done
[12:05:14.279] .../node_modules/@biomejs/biome postinstall: Done
[12:05:14.727] .../esbuild@0.21.5/node_modules/esbuild postinstall: Done
[12:05:14.950] .../sharp@0.33.5/node_modules/sharp install: Done
[12:05:15.159] 
[12:05:15.160] dependencies:
[12:05:15.160] + @astrojs/check 0.9.4
[12:05:15.160] + @astrojs/rss 4.0.9
[12:05:15.160] + @astrojs/sitemap 3.2.1
[12:05:15.166] + @astrojs/svelte 6.0.1
[12:05:15.166] + @astrojs/tailwind 5.1.2
[12:05:15.166] + @fontsource-variable/jetbrains-mono 5.1.1
[12:05:15.167] + @fontsource/roboto 5.1.0
[12:05:15.167] + @iconify-json/fa6-brands 1.2.1
[12:05:15.167] + @iconify-json/fa6-regular 1.2.1
[12:05:15.167] + @iconify-json/fa6-solid 1.2.1
[12:05:15.167] + @iconify-json/material-symbols 1.2.5
[12:05:15.167] + @iconify/svelte 4.0.2
[12:05:15.167] + @sveltejs/vite-plugin-svelte 4.0.0
[12:05:15.167] + @swup/astro 1.4.1
[12:05:15.168] + @tailwindcss/typography 0.5.15
[12:05:15.168] + astro 4.16.13
[12:05:15.168] + astro-compress 2.3.5
[12:05:15.168] + astro-icon 1.1.1
[12:05:15.168] + hastscript 9.0.0
[12:05:15.168] + katex 0.16.19
[12:05:15.168] + markdown-it 14.1.0
[12:05:15.168] + mdast-util-to-string 4.0.0
[12:05:15.168] + overlayscrollbars 2.10.0
[12:05:15.168] + pagefind 1.1.1
[12:05:15.168] + photoswipe 5.4.4
[12:05:15.168] + reading-time 1.5.0
[12:05:15.169] + rehype-autolink-headings 7.1.0
[12:05:15.169] + rehype-components 0.3.0
[12:05:15.169] + rehype-katex 7.0.1
[12:05:15.169] + rehype-slug 6.0.0
[12:05:15.169] + remark-directive 3.0.0
[12:05:15.169] + remark-directive-rehype 0.4.2
[12:05:15.169] + remark-github-admonitions-to-directives 1.0.5
[12:05:15.169] + remark-math 6.0.0
[12:05:15.169] + remark-sectionize 2.0.0
[12:05:15.169] + sanitize-html 2.13.1
[12:05:15.169] + sharp 0.33.5
[12:05:15.170] + stylus 0.63.0
[12:05:15.170] + svelte 5.2.2
[12:05:15.170] + tailwindcss 3.4.14
[12:05:15.170] + typescript 5.6.3
[12:05:15.170] + unist-util-visit 5.0.0
[12:05:15.170] 
[12:05:15.170] devDependencies:
[12:05:15.170] + @astrojs/ts-plugin 1.10.4
[12:05:15.170] + @biomejs/biome 1.8.3
[12:05:15.170] + @rollup/plugin-yaml 4.1.2
[12:05:15.170] + @types/markdown-it 14.1.2
[12:05:15.170] + @types/mdast 4.0.4
[12:05:15.170] + @types/sanitize-html 2.13.0
[12:05:15.171] + postcss-import 16.1.0
[12:05:15.171] + postcss-nesting 13.0.1
[12:05:15.171] 
[12:05:15.189] Done in 18.4s
[12:05:15.258] Running "pnpm run build"
[12:05:16.322] 
[12:05:16.322] > fuwari@0.0.1 build /vercel/path0
[12:05:16.323] > astro build && pagefind --site dist
[12:05:16.323] 
[12:05:19.075] 03:05:19 [vite] Forced re-optimization of dependencies
[12:05:19.701] 03:05:19 [types] Generated 683ms
[12:05:19.702] 03:05:19 [build] output: "static"
[12:05:19.702] 03:05:19 [build] directory: /vercel/path0/dist/
[12:05:19.702] 03:05:19 [build] Collecting build info...
[12:05:19.703] 03:05:19 [build] ✓ Completed in 795ms.
[12:05:19.705] 03:05:19 [build] Building static entrypoints...
[12:05:24.470] 03:05:24 [astro-icon] Loaded icons from fa6-brands, fa6-regular, fa6-solid, material-symbols
[12:05:26.874] 03:05:26 [vite] [32m✓ built in 7.13s[39m
[12:05:26.875] 03:05:26 [build] ✓ Completed in 7.17s.
[12:05:26.875] 
[12:05:26.876]  building client (vite) 
[12:05:26.895] 03:05:26 [vite] transforming...
[12:05:28.154] 03:05:28 [vite] [32m✓[39m 143 modules transformed.
[12:05:28.229] 03:05:28 [vite] rendering chunks...
[12:05:28.338] 03:05:28 [vite] computing gzip size...
[12:05:28.352] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[35mhoisted.DbgSiMEJ.css           [39m[1m[2m17.70 kB[22m[1m[22m[2m │ gzip:  3.70 kB[22m
[12:05:28.352] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mclass.Bvena8yX.js              [39m[1m[2m 0.16 kB[22m[1m[22m[2m │ gzip:  0.11 kB[22m
[12:05:28.352] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36murl-utils.Cvx6LZw3.js          [39m[1m[2m 0.26 kB[22m[1m[22m[2m │ gzip:  0.19 kB[22m
[12:05:28.352] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36minput.2AtaZ1Cd.js              [39m[1m[2m 0.44 kB[22m[1m[22m[2m │ gzip:  0.28 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36msetting-utils.De4sCiaH.js      [39m[1m[2m 0.91 kB[22m[1m[22m[2m │ gzip:  0.44 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mpreload-helper.CLcXU_4U.js     [39m[1m[2m 0.99 kB[22m[1m[22m[2m │ gzip:  0.60 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mclient.svelte.Bgw0o_X2.js      [39m[1m[2m 1.09 kB[22m[1m[22m[2m │ gzip:  0.60 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwupScriptsPlugin.o5PkFIdr.js  [39m[1m[2m 1.10 kB[22m[1m[22m[2m │ gzip:  0.61 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mindex.modern.CkIAsQri.js       [39m[1m[2m 1.76 kB[22m[1m[22m[2m │ gzip:  0.91 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mpage.D0rYE--H.js               [39m[1m[2m 1.87 kB[22m[1m[22m[2m │ gzip:  0.88 kB[22m
[12:05:28.353] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mDisplaySettings.RWvjpy_8.js    [39m[1m[2m 2.06 kB[22m[1m[22m[2m │ gzip:  1.11 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwupHeadPlugin.BKT_SVYP.js     [39m[1m[2m 2.52 kB[22m[1m[22m[2m │ gzip:  1.25 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mLightDarkSwitch.DLJ-ORTV.js    [39m[1m[2m 3.15 kB[22m[1m[22m[2m │ gzip:  1.27 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mzh_TW.DA4cTuyj.js              [39m[1m[2m 4.77 kB[22m[1m[22m[2m │ gzip:  2.05 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwupA11yPlugin.D84kWg1p.js     [39m[1m[2m 5.97 kB[22m[1m[22m[2m │ gzip:  2.29 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwupPreloadPlugin.CnFDLkk0.js  [39m[1m[2m 6.00 kB[22m[1m[22m[2m │ gzip:  2.35 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSearch.BAKrEkt5.js             [39m[1m[2m 6.11 kB[22m[1m[22m[2m │ gzip:  2.80 kB[22m
[12:05:28.354] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwupScrollPlugin.CPHDirUY.js   [39m[1m[2m 8.00 kB[22m[1m[22m[2m │ gzip:  2.40 kB[22m
[12:05:28.355] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mrender.CKYNopHH.js             [39m[1m[2m18.07 kB[22m[1m[22m[2m │ gzip:  7.32 kB[22m
[12:05:28.355] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mSwup.T76dPaas.js               [39m[1m[2m21.63 kB[22m[1m[22m[2m │ gzip:  7.41 kB[22m
[12:05:28.355] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mIcon.BB_x9DQs.js               [39m[1m[2m26.50 kB[22m[1m[22m[2m │ gzip: 10.78 kB[22m
[12:05:28.355] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mhoisted.pp1LUMP6.js            [39m[1m[2m54.07 kB[22m[1m[22m[2m │ gzip: 22.03 kB[22m
[12:05:28.355] 03:05:28 [vite] [2mdist/[22m[2m_astro/[22m[36mphotoswipe.esm.CKijkUPa.js     [39m[1m[2m60.42 kB[22m[1m[22m[2m │ gzip: 17.49 kB[22m
[12:05:28.355] 03:05:28 [vite] [32m✓ built in 1.46s[39m
[12:05:28.369] 
[12:05:28.369]  generating static routes 
[12:05:28.521] 03:05:28 ▶ src/pages/about.astro
[12:05:28.552] 03:05:28   └─ /about/index.html (+31ms)
[12:05:28.555] 03:05:28 ▶ src/pages/archive/category/uncategorized.astro
[12:05:28.561] 03:05:28   └─ /archive/category/uncategorized/index.html (+6ms)
[12:05:28.563] 03:05:28 ▶ src/pages/archive/category/[category].astro
[12:05:28.577] 03:05:28   ├─ /archive/category/Examples/index.html (+12ms)
[12:05:28.583] 03:05:28   └─ /archive/category/Guides/index.html (+7ms)
[12:05:28.585] 03:05:28 ▶ src/pages/archive/tag/[tag].astro
[12:05:28.594] 03:05:28   ├─ /archive/tag/Demo/index.html (+8ms)
[12:05:28.605] 03:05:28   ├─ /archive/tag/Example/index.html (+11ms)
[12:05:28.624] 03:05:28   ├─ /archive/tag/Markdown/index.html (+19ms)
[12:05:28.630] 03:05:28   ├─ /archive/tag/Fuwari/index.html (+7ms)
[12:05:28.637] 03:05:28   ├─ /archive/tag/Blogging/index.html (+6ms)
[12:05:28.647] 03:05:28   ├─ /archive/tag/Customization/index.html (+11ms)
[12:05:28.658] 03:05:28   └─ /archive/tag/Video/index.html (+10ms)
[12:05:28.659] 03:05:28 ▶ src/pages/archive/index.astro
[12:05:28.663] 03:05:28   └─ /archive/index.html (+4ms)
[12:05:28.674] 03:05:28 ▶ src/pages/posts/[...slug].astro
[12:05:28.687] 03:05:28   ├─ /posts/guide/index.html (+11ms)
[12:05:28.695] 03:05:28   ├─ /posts/markdown-extended/index.html (+9ms)
[12:05:28.706] 03:05:28   ├─ /posts/markdown/index.html (+11ms)
[12:05:28.714] 03:05:28   └─ /posts/video/index.html (+8ms)
[12:05:28.715] 03:05:28 λ src/pages/robots.txt.ts
[12:05:28.717] 03:05:28   └─ /robots.txt (+1ms)
[12:05:28.772] 03:05:28 λ src/pages/rss.xml.ts
[12:05:28.807] 03:05:28   └─ /rss.xml (+34ms)
[12:05:28.810] 03:05:28 ▶ src/pages/[...page].astro
[12:05:28.820] 03:05:28   └─ /index.html (+9ms)
[12:05:28.820] 03:05:28 ✓ Completed in 453ms.
[12:05:28.820] 
[12:05:28.821]  generating optimized images 
[12:05:28.898] 03:05:28   ▶ /_astro/demo-avatar.CxcI0ivM_ZCdd5H.webp (before: 406kB, after: 41kB) (+77ms) (1/2)
[12:05:29.125] 03:05:29   ▶ /_astro/cover.CgGywNHJ_1wRFtk.webp (before: 218kB, after: 144kB) (+226ms) (2/2)
[12:05:29.127] 03:05:29 ✓ Completed in 306ms.
[12:05:29.127] 
[12:05:29.144] 03:05:29 [@astrojs/sitemap] `sitemap-index.xml` created at `dist`
[12:05:29.588] (-6.02 KB)	10.10% reduction in //dist/index.html
[12:05:29.763] (-4.23 KB)	10.98% reduction in //dist/about/index.html
[12:05:29.917] (-5.23 KB)	12.63% reduction in //dist/archive/index.html
[12:05:30.104] (-4.69 KB)	9.23% reduction in //dist/posts/guide/index.html
[12:05:30.308] (-5.04 KB)	8.64% reduction in //dist/posts/markdown/index.html
[12:05:30.427] (-5.21 KB)	9.42% reduction in //dist/posts/markdown-extended/index.html
[12:05:30.566] (-4.53 KB)	9.51% reduction in //dist/posts/video/index.html
[12:05:30.679] (-4.84 KB)	12.11% reduction in //dist/archive/category/Examples/index.html
[12:05:30.778] (-4.05 KB)	11.02% reduction in //dist/archive/category/Guides/index.html
[12:05:30.867] (-3.64 KB)	10.42% reduction in //dist/archive/category/uncategorized/index.html
[12:05:30.962] (-4.45 KB)	11.55% reduction in //dist/archive/tag/Blogging/index.html
[12:05:31.041] (-4.05 KB)	11.02% reduction in //dist/archive/tag/Customization/index.html
[12:05:31.138] (-4.45 KB)	11.55% reduction in //dist/archive/tag/Demo/index.html
[12:05:31.212] (-4.45 KB)	11.54% reduction in //dist/archive/tag/Example/index.html
[12:05:31.292] (-4.44 KB)	11.64% reduction in //dist/archive/tag/Fuwari/index.html
[12:05:31.364] (-4.45 KB)	11.54% reduction in //dist/archive/tag/Markdown/index.html
[12:05:31.437] (-4.05 KB)	11.03% reduction in //dist/archive/tag/Video/index.html
[12:05:31.439] ✓ Successfully compressed a total of 17 HTML files for 77.83 KB.
[12:05:31.459] (-23 Bytes)	1.12% reduction in //dist/_astro/DisplaySettings.RWvjpy_8.js
[12:05:31.816] (-59 Bytes)	0.22% reduction in //dist/_astro/Icon.BB_x9DQs.js
[12:05:31.832] (-NaN undefined)	-0.10% reduction in //dist/_astro/LightDarkSwitch.DLJ-ORTV.js
[12:05:31.889] (-79 Bytes)	1.29% reduction in //dist/_astro/Search.BAKrEkt5.js
[12:05:32.169] 03:05:32 [build] Waiting for integration "astro-compress", hook "astro:build:done"...
[12:05:32.172] (-NaN undefined)	-0.43% reduction in //dist/_astro/Swup.T76dPaas.js
[12:05:32.229] (-NaN undefined)	-0.02% reduction in //dist/_astro/SwupA11yPlugin.D84kWg1p.js
[12:05:32.252] (-NaN undefined)	-0.71% reduction in //dist/_astro/SwupHeadPlugin.BKT_SVYP.js
[12:05:32.295] (-NaN undefined)	-0.60% reduction in //dist/_astro/SwupPreloadPlugin.CnFDLkk0.js
[12:05:32.304] (-NaN undefined)	-0.09% reduction in //dist/_astro/SwupScriptsPlugin.o5PkFIdr.js
[12:05:32.370] (-NaN undefined)	-0.11% reduction in //dist/_astro/SwupScrollPlugin.CPHDirUY.js
[12:05:32.373] (-1 Bytes)	0.64% reduction in //dist/_astro/class.Bvena8yX.js
[12:05:32.381] (-NaN undefined)	-0.83% reduction in //dist/_astro/client.svelte.Bgw0o_X2.js
[12:05:32.851] (-425 Bytes)	0.79% reduction in //dist/_astro/hoisted.pp1LUMP6.js
[12:05:32.895] (-NaN undefined)	-0.34% reduction in //dist/_astro/index.modern.CkIAsQri.js
[12:05:32.900] (-10 Bytes)	2.27% reduction in //dist/_astro/input.2AtaZ1Cd.js
[12:05:32.916] (-NaN undefined)	-1.93% reduction in //dist/_astro/page.D0rYE--H.js
[12:05:33.215] (-25 Bytes)	0.04% reduction in //dist/_astro/photoswipe.esm.CKijkUPa.js
[12:05:33.222] (-NaN undefined)	-1.02% reduction in //dist/_astro/preload-helper.CLcXU_4U.js
[12:05:33.404] (-150 Bytes)	0.83% reduction in //dist/_astro/render.CKYNopHH.js
[12:05:33.410] (-12 Bytes)	1.32% reduction in //dist/_astro/setting-utils.De4sCiaH.js
[12:05:33.414] (-15 Bytes)	5.84% reduction in //dist/_astro/url-utils.Cvx6LZw3.js
[12:05:33.434] (-1 Bytes)	0.02% reduction in //dist/_astro/zh_TW.DA4cTuyj.js
[12:05:33.435] ✓ Successfully compressed a total of 22 JavaScript files for 577 Bytes.
[12:05:33.438] 03:05:33 [build] 17 page(s) built in 14.53s
[12:05:33.439] 03:05:33 [build] Complete!
[12:05:33.550] 
[12:05:33.551] Running Pagefind v1.1.1 (Extended)
[12:05:33.551] Running from: "/vercel/path0"
[12:05:33.551] Source:       "dist"
[12:05:33.551] Output:       "dist/pagefind"
[12:05:33.552] 
[12:05:33.552] [Walking source directory]
[12:05:33.553] Found 17 files matching **/*.{html}
[12:05:33.553] 
[12:05:33.554] [Parsing files]
[12:05:33.605] Found a data-pagefind-body element on the site.
[12:05:33.605] ↳ Ignoring pages without this tag.
[12:05:33.606] 
[12:05:33.606] [Reading languages]
[12:05:33.606] Discovered 1 language: en
[12:05:33.606] 
[12:05:33.606] [Building search indexes]
[12:05:33.607] Total: 
[12:05:33.607]   Indexed 1 language
[12:05:33.608]   Indexed 5 pages
[12:05:33.608]   Indexed 471 words
[12:05:33.608]   Indexed 0 filters
[12:05:33.608]   Indexed 0 sorts
[12:05:33.614] 
[12:05:33.614] Finished in 0.065 seconds
[12:05:33.726] Build Completed in /vercel/output [39s]
[12:05:33.876] Deploying outputs...
[12:05:35.952] 
[12:05:36.249] Deployment completed
[12:05:51.207] Uploading build cache [128.50 MB]...
[12:05:52.889] Build cache uploaded: 1.683s
```

Hasenpfote commented:
I've taken an interest in [this log](https://github.com/saicaca/fuwari/pull/283#issuecomment-2586879034) of PR #283, so I looked into the currently open PRs.
After examining the `package.json` files in each individual fork, it seems that there are no significant differences in the pnpm versions.
Hmm, what could it be?

Here are the details from my fork:
https://github.com/Hasenpfote/fuwari/blob/48b9c137e866e19dd3b0505b32674eba67e6ef69/package.json#L71

Additionally, here is the actual version installed on my local PC, not in my Vercel environment.
Node.js version: 22.13.0
pnpm version: 9.14.4

saicaca commented:
```
04:24:52 ▶ src/pages/about.astro
04:24:52   └─ /about/index.html
aboutPost.render is not a function
```

I think I know how this error occurred. Vercel cached an object built in an Astro 5.0 branch and somehow used it while building the 4.x branches. Astro 4.x couldn't process the object, leading to a build failure.

I’ll update the main branch to 5.0 now, and that should fix the problem.

Hasenpfote commented:
This is in a different environment than the current issue, but I occasionally experience an error on my local PC and Netlify.
These errors were resolved by a rebuild, so I had forgotten about them.
I'm not sure if it's related, but I'll include the log just in case.

The problematic part is here:
```shell
03:55:18 [ERROR] [vite] x Build failed in 4.80s
```

Full log:
```shell
12:54:35 PM: build-image version: bb0c0791f5fe3f33dad765dcb71d506f5ea22499 (noble)
12:54:35 PM: buildbot version: a980cf9efb6f102593227183a239d2758329d6aa
12:54:35 PM: Building without cache
12:54:35 PM: Starting to prepare the repo for build
12:54:36 PM: No cached dependencies found. Cloning fresh repo
12:54:36 PM: git clone --filter=blob:none https://github.com/Hasenpfote/astro-blog-2
12:54:36 PM: Preparing Git Reference refs/heads/main
12:54:38 PM: Starting to install dependencies
12:54:38 PM: Python version set to 3.13.1
12:54:38 PM: Ruby version set to 2.7.2
12:54:39 PM: mise go@1.19         install
12:54:39 PM: mise go@1.19         download go1.19.linux-amd64.tar.gz
12:54:40 PM: mise go@1.19         checksum go1.19.linux-amd64.tar.gz
12:54:40 PM: mise go@1.19         extract go1.19.linux-amd64.tar.gz
12:54:42 PM: mise go@1.19         go version
12:54:42 PM: mise go@1.19         go version go1.19 linux/amd64
12:54:42 PM: mise go@1.19       ✓ installed
12:54:42 PM: Go version set to 1.19
12:54:42 PM: Using PHP version 8.3
12:54:43 PM: v18.20.5 is already installed.
12:54:44 PM: Now using node v18.20.5 (npm v10.8.2)
12:54:44 PM: Enabling Node.js Corepack
12:54:44 PM: Started restoring cached build plugins
12:54:44 PM: Finished restoring cached build plugins
12:54:44 PM: Started restoring cached corepack dependencies
12:54:44 PM: Finished restoring cached corepack dependencies
12:54:45 PM: No pnpm workspaces detected
12:54:45 PM: Started restoring cached node modules
12:54:45 PM: Finished restoring cached node modules
12:54:45 PM: Installing npm packages using pnpm version 9.14.4
12:54:46 PM: Lockfile is up to date, resolution step is skipped
12:54:46 PM: Progress: resolved 1, reused 0, downloaded 0, added 0
12:54:47 PM: Packages: +1152
12:54:47 PM: ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
12:54:47 PM: Progress: resolved 1152, reused 0, downloaded 0, added 0
12:54:48 PM: Progress: resolved 1152, reused 0, downloaded 159, added 137
12:54:49 PM: Progress: resolved 1152, reused 0, downloaded 550, added 550
12:54:50 PM: Progress: resolved 1152, reused 0, downloaded 970, added 969
12:54:51 PM: Progress: resolved 1152, reused 0, downloaded 1145, added 1145
12:54:52 PM: Progress: resolved 1152, reused 0, downloaded 1151, added 1151
12:54:52 PM: Progress: resolved 1152, reused 0, downloaded 1152, added 1152, done
12:54:52 PM: .../.pnpm/swup@4.8.1/node_modules/swup postinstall$ opencollective-postinstall || true
12:54:52 PM: .../node_modules/@biomejs/biome postinstall$ node scripts/postinstall.js
12:54:52 PM: .../sharp@0.33.5/node_modules/sharp install$ node install/check
12:54:52 PM: .../esbuild@0.21.5/node_modules/esbuild postinstall$ node install.js
12:54:52 PM: .../.pnpm/swup@4.8.1/node_modules/swup postinstall: Done
12:54:52 PM: .../node_modules/@biomejs/biome postinstall: Done
12:54:52 PM: .../esbuild@0.21.5/node_modules/esbuild postinstall: Done
12:54:56 PM: .../sharp@0.33.5/node_modules/sharp install: Done
12:54:57 PM: dependencies:
12:54:57 PM: + @astrojs/check 0.9.4
12:54:57 PM: + @astrojs/rss 4.0.9
12:54:57 PM: + @astrojs/sitemap 3.2.1
12:54:57 PM: + @astrojs/svelte 6.0.1
12:54:57 PM: + @astrojs/tailwind 5.1.2
12:54:57 PM: + @fontsource-variable/jetbrains-mono 5.1.1
12:54:57 PM: + @fontsource/roboto 5.1.0
12:54:57 PM: + @iconify-json/fa6-brands 1.2.1
12:54:57 PM: + @iconify-json/fa6-regular 1.2.1
12:54:57 PM: + @iconify-json/fa6-solid 1.2.1
12:54:57 PM: + @iconify-json/material-symbols 1.2.5
12:54:57 PM: + @iconify/svelte 4.0.2
12:54:57 PM: + @sveltejs/vite-plugin-svelte 4.0.0
12:54:57 PM: + @swup/astro 1.4.1
12:54:57 PM: + @tailwindcss/typography 0.5.15
12:54:57 PM: + astro 4.16.13
12:54:57 PM: + astro-compress 2.3.5
12:54:57 PM: + astro-icon 1.1.1
12:54:57 PM: + hastscript 9.0.0
12:54:57 PM: + markdown-it 14.1.0
12:54:57 PM: + mdast-util-to-string 4.0.0
12:54:57 PM: + overlayscrollbars 2.10.0
12:54:57 PM: + pagefind 1.1.1
12:54:57 PM: + photoswipe 5.4.4
12:54:57 PM: + playwright 1.49.1
12:54:57 PM: + reading-time 1.5.0
12:54:57 PM: + rehype-autolink-headings 7.1.0
12:54:57 PM: + rehype-components 0.3.0
12:54:57 PM: + rehype-external-links 3.0.0
12:54:57 PM: + rehype-katex 7.0.1
12:54:57 PM: + rehype-mermaid 3.0.0
12:54:57 PM: + rehype-slug 6.0.0
12:54:57 PM: + remark-directive 3.0.0
12:54:57 PM: + remark-directive-rehype 0.4.2
12:54:57 PM: + remark-github-admonitions-to-directives 1.0.5
12:54:57 PM: + remark-math 6.0.0
12:54:57 PM: + remark-sectionize 2.0.0
12:54:57 PM: + sanitize-html 2.13.1
12:54:57 PM: + sharp 0.33.5
12:54:57 PM: + stylus 0.63.0
12:54:57 PM: + svelte 5.2.2
12:54:57 PM: + tailwindcss 3.4.14
12:54:57 PM: + typescript 5.6.3
12:54:57 PM: + unist-util-visit 5.0.0
12:54:57 PM: devDependencies:
12:54:57 PM: + @astrojs/ts-plugin 1.10.4
12:54:57 PM: + @biomejs/biome 1.8.3
12:54:57 PM: + @rollup/plugin-yaml 4.1.2
12:54:57 PM: + @types/markdown-it 14.1.2
12:54:57 PM: + @types/mdast 4.0.4
12:54:57 PM: + @types/sanitize-html 2.13.0
12:54:57 PM: + postcss-import 16.1.0
12:54:57 PM: + postcss-nesting 13.0.1
12:54:57 PM: > fuwari@0.0.1 postinstall /opt/build/repo
12:54:57 PM: > pnpm dlx playwright-core install chromium
12:54:57 PM: Progress: resolved 1, reused 0, downloaded 0, added 0
12:54:57 PM: Packages: +1
12:54:57 PM: +
12:54:57 PM: Progress: resolved 1, reused 1, downloaded 0, added 1, done
12:54:58 PM: Downloading Chromium 131.0.6778.33 (playwright build v1148) from https://playwright.azureedge.net/builds/chromium/1148/chromium-linux.zip
12:54:58 PM: |                                                                                |   0% of 161.3 MiB
12:54:58 PM: |■■■■■■■■                                                                        |  10% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■                                                                |  20% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 161.3 MiB
12:54:58 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 161.3 MiB
12:54:59 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 161.3 MiB
12:54:59 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 161.3 MiB
12:54:59 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 161.3 MiB
12:55:03 PM: Chromium 131.0.6778.33 (playwright build v1148) downloaded to /opt/buildhome/.cache/ms-playwright/chromium-1148
12:55:03 PM: Downloading FFMPEG playwright build v1010 from https://playwright.azureedge.net/builds/ffmpeg/1010/ffmpeg-linux.zip
12:55:03 PM: |                                                                                |   0% of 2.3 MiB
12:55:03 PM: |■■■■■■■■                                                                        |  10% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■                                                                |  20% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 2.3 MiB
12:55:03 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 2.3 MiB
12:55:03 PM: FFMPEG playwright build v1010 downloaded to /opt/buildhome/.cache/ms-playwright/ffmpeg-1010
12:55:03 PM: Downloading Chromium Headless Shell 131.0.6778.33 (playwright build v1148) from https://playwright.azureedge.net/builds/chromium/1148/chromium-headless-shell-linux.zip
12:55:03 PM: |                                                                                |   0% of 100.9 MiB
12:55:04 PM: |■■■■■■■■                                                                        |  10% of 100.9 MiB
12:55:04 PM: |■■■■■■■■■■■■■■■■                                                                |  20% of 100.9 MiB
12:55:04 PM: |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 100.9 MiB
12:55:04 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 100.9 MiB
12:55:04 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 100.9 MiB
12:55:05 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 100.9 MiB
12:55:05 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 100.9 MiB
12:55:05 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 100.9 MiB
12:55:05 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 100.9 MiB
12:55:05 PM: |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 100.9 MiB
12:55:07 PM: Chromium Headless Shell 131.0.6778.33 (playwright build v1148) downloaded to /opt/buildhome/.cache/ms-playwright/chromium_headless_shell-1148
12:55:08 PM: Done in 22.2s
12:55:08 PM: npm packages installed using pnpm
12:55:08 PM: Successfully installed dependencies
12:55:08 PM: Starting build script
12:55:09 PM: Detected 2 framework(s)
12:55:09 PM: "astro" at version "4.16.13"
12:55:09 PM: "svelte" at version "5.2.2"
12:55:09 PM: Section completed: initializing
12:55:10 PM: ​
12:55:10 PM: Netlify Build                                                 
12:55:10 PM: ────────────────────────────────────────────────────────────────
12:55:10 PM: ​
12:55:10 PM: ❯ Version
12:55:10 PM:   @netlify/build 29.58.3
12:55:10 PM: ​
12:55:10 PM: ❯ Flags
12:55:10 PM:   accountId: 5fe54e1c500f908f09c3654c
12:55:10 PM:   baseRelDir: true
12:55:10 PM:   buildId: 6789d46f09439aa2f5640722
12:55:10 PM:   deployId: 6789d46f09439aa2f5640724
12:55:10 PM: ​
12:55:10 PM: ❯ Current directory
12:55:10 PM:   /opt/build/repo
12:55:10 PM: ​
12:55:10 PM: ❯ Config file
12:55:10 PM:   /opt/build/repo/netlify.toml
12:55:10 PM: ​
12:55:10 PM: ❯ Context
12:55:10 PM:   production
12:55:10 PM: ​
12:55:10 PM: Build command from Netlify app                                
12:55:10 PM: ────────────────────────────────────────────────────────────────
12:55:10 PM: ​
12:55:10 PM: $ pnpm run build
12:55:11 PM: > fuwari@0.0.1 build /opt/build/repo
12:55:11 PM: > astro build && pagefind --site dist
12:55:12 PM: 03:55:12 [vite] Forced re-optimization of dependencies
12:55:13 PM: 03:55:13 [types] Generated 458ms
12:55:13 PM: 03:55:13 [build] output: "static"
12:55:13 PM: 03:55:13 [build] directory: /opt/build/repo/dist/
12:55:13 PM: 03:55:13 [build] Collecting build info...
12:55:13 PM: 03:55:13 [build] ✓ Completed in 504ms.
12:55:13 PM: 03:55:13 [build] Building static entrypoints...
12:55:18 PM: [Shiki] The language "css{:title=stylesheet.css" doesn't exist, falling back to "plaintext".
[Shiki] The language "javascript{:title=gatsby-config.js" doesn't exist, falling back to "plaintext".
[Shiki] The language "{:title=subset.txt" doesn't exist, falling back to "plaintext".
[Shiki] The language "sh{:title=autorun.sh" doesn't exist, falling back to "plaintext".
[Shiki] The language "txt{:title=qpkg.conf" doesn't exist, falling back to "plaintext".
[Shiki] The language "txt{:title=sudoers" doesn't exist, falling back to "plaintext".
[Shiki] The language "txt{:title=.profile" doesn't exist, falling back to "plaintext".
03:55:18 [ERROR] [vite] x Build failed in 4.80s
12:55:19 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
12:55:19 PM: [vite:css] [postcss] /opt/build/repo/src/styles/markdown.css:23:9: The `link` class does not exist. If `link` is a custom class, make sure it is defined within a `@layer` directive.
12:55:19 PM: file: /opt/build/repo/src/styles/markdown.css:23:8
12:55:19 PM:   Location:
12:55:19 PM:     /opt/build/repo/src/styles/markdown.css:23:8
12:55:19 PM:   Stack trace:
12:55:19 PM:     at Input.error (/opt/build/repo/node_modules/.pnpm/postcss@8.4.47/node_modules/postcss/lib/input.js:106:16)
12:55:19 PM:     at processApply (/opt/build/repo/node_modules/.pnpm/tailwindcss@3.4.14/node_modules/tailwindcss/lib/lib/expandApplyAtRules.js:380:29)
12:55:19 PM:     at /opt/build/repo/node_modules/.pnpm/tailwindcss@3.4.14/node_modules/tailwindcss/lib/processTailwindFeatures.js:55:50
12:55:19 PM:     at async LazyResult.runAsync (/opt/build/repo/node_modules/.pnpm/postcss@8.4.47/node_modules/postcss/lib/lazy-result.js:261:11)
12:55:19 PM:     at async Object.transform (file:///opt/build/repo/node_modules/.pnpm/vite@5.4.10_@types+node@22.8.2_lightningcss@1.27.0_sass@1.80.4_stylus@0.63.0_terser@5.36.0/node_modules/vite/dist/node/chunks/dep-BWSbWtLw.js:36170:11)
12:55:19 PM:  ELIFECYCLE  Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
12:55:19 PM: ​
12:55:19 PM: "build.command" failed                                        
12:55:19 PM: ────────────────────────────────────────────────────────────────
12:55:19 PM: ​
12:55:19 PM:   Error message
12:55:19 PM:   Command failed with exit code 1: pnpm run build (https://ntl.fyi/exit-code-1)
12:55:19 PM: ​
12:55:19 PM:   Error location
12:55:19 PM:   In Build command from Netlify app:
12:55:19 PM:   pnpm run build
12:55:19 PM: ​
12:55:19 PM:   Resolved config
12:55:19 PM:   build:
12:55:19 PM:     command: pnpm run build
12:55:19 PM:     commandOrigin: ui
12:55:19 PM:     publish: /opt/build/repo/dist
12:55:19 PM:     publishOrigin: ui
12:55:19 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
12:55:19 PM: Failing build: Failed to build site
12:55:20 PM: Finished processing build request in 44.539s
```

saicaca commented:
> This is in a different environment than the current issue, but I occasionally experience an error on my local PC and Netlify. These errors were resolved by a rebuild, so I had forgotten about them. I'm not sure if it's related, but I'll include the log just in case.

It seems to be a CSS issue and not related to the changes here. I'll take another look later if this issue persists. Let's merge this first. Thank you for your contribution!
